core/services/ocr2/delegate.go

```diff
@@ -305,6 +316,51 @@ func (d *Delegate) cleanupEVM(ctx context.Context, jb job.Job, relayID types.Rel
 			d.lggr.Errorw("failed to derive ocr2keeper filter names from spec", "err", err, "spec", spec)
 		}
 		filters = append(filters, filters21...)
+	case types.CCIPCommit:
+		// Write PluginConfig bytes to send source/dest relayer provider + info outside of top level rargs/pargs over the wire
+		var pluginJobSpecConfig ccipconfig.CommitPluginJobSpecConfig
+		err = json.Unmarshal(spec.PluginConfig.Bytes(), &pluginJobSpecConfig)
+		if err != nil {
+			return err
+		}
+
+		dstProvider, err2 := d.ccipCommitGetDstProvider(ctx, jb, pluginJobSpecConfig, transmitterID)
+		if err2 != nil {
+			return err
+		}
+
+		srcProvider, _, err2 := d.ccipCommitGetSrcProvider(ctx, jb, pluginJobSpecConfig, transmitterID, dstProvider)
+		if err2 != nil {
+			return err
+		}
+		err2 = ccipcommit.UnregisterCommitPluginLpFilters(srcProvider, dstProvider)
+		if err2 != nil {
+			d.lggr.Errorw("failed to unregister ccip commit plugin filters", "err", err2, "spec", spec)
+		}
+		return nil
+	case types.CCIPExecution:
+		// PROVIDER BASED ARG CONSTRUCTION
+		// Write PluginConfig bytes to send source/dest relayer provider + info outside of top level rargs/pargs over the wire
+		var pluginJobSpecConfig ccipconfig.ExecPluginJobSpecConfig
+		err = json.Unmarshal(spec.PluginConfig.Bytes(), &pluginJobSpecConfig)
+		if err != nil {
+			return err
+		}
+
+		dstProvider, err2 := d.ccipExecGetDstProvider(ctx, jb, pluginJobSpecConfig, transmitterID)
+		if err2 != nil {
+			return err
+		}
+
+		srcProvider, _, err2 := d.ccipExecGetSrcProvider(ctx, jb, pluginJobSpecConfig, transmitterID, dstProvider)
+		if err2 != nil {
+			return err
+		}
+		err2 = ccipexec.UnregisterExecPluginLpFilters(srcProvider, dstProvider)
+		if err2 != nil {
+			d.lggr.Errorw("failed to unregister ccip exec plugin filters", "err", err2, "spec", spec)
+		}
+		return nil
```

Comment:
Why does this logic live in `cleanupEVM` as opposed to triggering when a service is closed?